### PR TITLE
fix: Modal footer buttons not spaced when using href in button

### PR DESCRIPTION
**Pull request by headwindz**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

[Modal footer buttons not spaced when using href in button](https://github.com/ant-design/ant-design/issues/29658)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  fix the bug that Modal footer buttons not spaced properly when using href in button   |
| 🇨🇳 Chinese |  修复Modal页脚里使用href按钮导致的间距丢失问题  |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by n0ruSh** [comment]
@afc163 @hengkx  Do you have any ideas about why the `Preview Build / build preview(pull_request)` task fails? I run `npm run site` locally and it seems to be fine.

**Review by inottn — COMMENTED** [review]
supplement [rtl](https://github.com/ant-design/ant-design/blob/be299cbe708d7439ddd333759f4f9a4b44b3065a/components/modal/style/rtl.less#L57) ?

File: components/modal/style/modal.less [hunk 1]
```diff
@@ -94,7 +94,8 @@
       @modal-footer-border-color-split;
     border-radius: 0 0 @border-radius-base @border-radius-base;
 
-    button + button {
+    button:not(:first-child),
+    a:not(:first-child) {
```
Comment (afc163) [inline components/modal/style/modal.less:98]:
```suggestion
    .@{ant-prefix}-btn + .@{ant-prefix}-btn {
```

Comment (afc163) [reply components/modal/style/modal.less:98]: Or use Space

**Pull request closed without merging**